xbmc/filesystem/SMBFile.cpp

```diff
@@ -151,6 +158,10 @@ void CSMB::Init()
 
     // setup our context
     m_context = smbc_new_context();
+
+    // restore HOME
+    setenv("HOME", truehome, 1);
```

Comment:
Apparently this call *is* succeeding (return value is 0) however `errno` is being set to 95 which (according to `strerror(errno)`) means ` Operation not supported`, so it doesn't look like the $HOME variable is being reset to the original value.

Any ideas?